test/contrib/aws/instrumentation_test.rb

```diff
@@ -0,0 +1,68 @@
+require 'helper'
+require 'aws-sdk'
+require 'ddtrace'
+require 'ddtrace/contrib/aws/patcher'
+require 'ddtrace/ext/http'
+
+module Datadog
+  module Contrib
+    module Aws
+      class InstrumentationTest < Minitest::Test
+        def setup
+          # initializes the tracer and a stub client
+          @tracer = enable_tracer
+          @client = ::Aws::S3::Client.new(stub_responses: true)
+        end
+
+        def test_pin_defaults
+          assert_equal('aws', pin.service)
+          assert_equal('web', pin.app_type)
+          assert_nil(pin.app)
+          assert_nil(pin.name)
+          assert_nil(pin.config)
+        end
+
+        def test_list_buckets
```

Comment:
@p-lambert it's enough to have all the checks in one test since it should reproduce a real use case (i.e. a call to `list_buckets` or a more complex scenario that reproduces/protects from a regression).